packages/ethereum/contracts/HoprNetworkRegistry.sol

```diff
@@ -75,22 +89,52 @@ contract HoprNetworkRegistry is Ownable {
    * @notice It allows msg.sender to update registered node address.
    * @param hoprAddress Hopr nodes id. e.g. 16Uiu2HAmHsB2c2puugVuuErRzLm9NZfceainZpkxqJMR6qGsf1x1
    */
-  function selfRegister(string calldata hoprAddress) external mustBeEnabled returns (bool) {
+  function selfRegister(string memory hoprAddress) external mustBeEnabled returns (bool) {
+    require(bytes(hoprAddress).length > 0, 'HoprNetworkRegistry: HOPR node multiaddress must not be empty');
+    // get account associated with the given hopr node address, if any
+    address registeredAccount = nodeMultiaddrToAccount[hoprAddress];
+    // if the hopr node multiaddress was linked to a different account, revert.
+    // To change a nodes' linked account, it must be deregistered by the previously linked account
+    // first before registering by the new account, to prevent hostile takeover of others' node address
+    require(
+      registeredAccount == msg.sender || registeredAccount == address(0),
+      'HoprNetworkRegistry: Cannot link a registered node to a different account'
+    );
+
+    // get multi address associated with the caller, if any
+    bytes memory registeredNodeMultiaddrInBytes = bytes(accountToNodeMultiaddr[msg.sender]);
+    require(
+      registeredNodeMultiaddrInBytes.length == 0 ||
+        keccak256(registeredNodeMultiaddrInBytes) == keccak256(bytes(hoprAddress)),
+      'HoprNetworkRegistry: Cannot link an account to a different node. Please remove the registered node'
+    );
+
     if (requirementImplementation.isRequirementFulfilled(msg.sender)) {
-      accountToNodeMultiaddr[msg.sender] = hoprAddress;
-      nodeMultiaddrToAccount[hoprAddress] = msg.sender;
-      emit Registered(msg.sender, hoprAddress);
+      // only update the list when no record previously exists
+      if (registeredNodeMultiaddrInBytes.length == 0) {
+        accountToNodeMultiaddr[msg.sender] = hoprAddress;
+        nodeMultiaddrToAccount[hoprAddress] = msg.sender;
+        emit Registered(msg.sender, hoprAddress);
+      }
       emit EligibilityUpdated(msg.sender, true);
```

Comment:
Should this be emitted even though we didn't update anything?

Reply:
No need to do so. If the link between the`msg.sender` and `hoprAddress` remains the same, there should have been an event emitted, so skip the emission of `Registered` event.